src/modules/fancyzones/FancyZonesLib/ZoneWindow.cpp

```diff
@@ -140,19 +140,22 @@ struct ZoneWindow : public winrt::implements<ZoneWindow, IZoneWindow>
     ClearSelectedZones() noexcept;
     IFACEMETHODIMP_(void)
     FlashZones() noexcept;
+    IFACEMETHODIMP_(void)
+    SetZoneColors(const ZoneColors& colors) noexcept;
+    IFACEMETHODIMP_(void)
+    SetOverlappingZonesAlgorithm(OverlappingZonesAlgorithm overlappingAlgorithm) noexcept;
 
 protected:
     static LRESULT CALLBACK s_WndProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam) noexcept;
 
 private:
     void InitializeZoneSets(const std::wstring& parentUniqueId) noexcept;
-    void CalculateZoneSet() noexcept;
+    void CalculateZoneSet(OverlappingZonesAlgorithm overlappingAlgorithm) noexcept;
```

Comment:
Why passing `overlappingAlgorithm` as arg?  `m_overlappingAlgorithm` can be used directly in `CalculateZoneSet`. Or am I missing something?

Reply:
It can be used directly, but IMO it feels like `OverlappingZonesAlgorithm` is not the thing that should be in the `ZoneWindow` class. I would prefer to extract it along with `ZoneColors` later.